lib/lib.sh

```diff
@@ -340,14 +343,18 @@ setup_interface() {
     while [ "$(date +%s)" -lt $deadline ]; do
         local -i changes=0
         local -i ifid
-        ifid=$(echo $iface | tr -d a-z)
+        ifid=$(echo $iface | tr -d a-zA-Z)
         [ -n "$ifid" ] || { error "Unable to get index of $iface" ; exit 2; }
         mkdir -p /run/network/$iface
         echo $ifid > /run/network/$iface/pref
 
         changes+=$(create_interface_config "$iface" "$ifid" "$ether")
         for family in 4 6; do
-            changes+=$(create_rules "$iface" "$ifid" $family)
+            if [ "$device_number" -eq 0 ] && [ "$ether" = "$default_mac" ]; then
+                debug "Skipping ipv$family rules for default ENI $iface $ether $default_mac $device_number"
```

Comment:
Thinking about this a little more, I realized that we're still creating the [extra routing tables](https://github.com/amazonlinux/amazon-ec2-net-utils/blob/507a1184b23fdf9163dfc2001118422a12955b2b/lib/lib.sh#L243-L258) in the `create_if_overrides` function. I don't think this is a blocker, because the important thing here is that we're skipping rule creation and that the `main` table has everything we need.

This should be revisited in the future though, as we shouldn't be creating unused resources like this.